Submitted By:     Adam Zlehovszky <zleho@index.hu>
Date:             2004.02.09.
Initial Packages: 2.9.5
Upstream Status: Not submitted - Frugalware specific
Description:      This patch fixes an annoying bug of pacman.
                  Always extract config files as .pacnew, instead of overwriting
                  the users's configs.

--- pacman-2.9.5.orig/src/pacman.c	2005-01-12 00:14:16.000000000 +0100
+++ pacman-2.9.5/src/pacman.c	2005-02-19 13:30:43.989002328 +0100
@@ -1885,13 +1885,13 @@
 						installnew = 1;
 					} else {
 						char newpath[PATH_MAX];
-						vprint("  action: saving current file and installing new one\n");
-						installnew = 1;
-						snprintf(newpath, PATH_MAX, "%s.pacsave", expath);
-						if(rename(expath, newpath)) {
-							logaction(stderr, "error: could not rename %s: %s", expath, strerror(errno));
+						vprint("  action: keeping current file and installing new one with .pacnew ending\n");
+						installnew = 0;
+						snprintf(newpath, PATH_MAX, "%s.pacnew", expath);
+						if(copyfile(temp, newpath)) {
+							logaction(stderr, "error: could not install %s as %s: %s", expath, newpath, strerror(errno));
 						} else {
-							logaction(stderr, "warning: %s saved as %s", expath, newpath);
+							logaction(stderr, "warning: %s installed as %s", expath, newpath);
 						}
 					}
 
diff -Naur pacman-2.9.6.orig/doc/pacman.8.in pacman-2.9.6/doc/pacman.8.in
--- pacman-2.9.6.orig/doc/pacman.8.in	2005-03-06 20:39:38.000000000 +0100
+++ pacman-2.9.6/doc/pacman.8.in	2005-08-20 14:25:02.000000000 +0200
@@ -186,7 +186,7 @@
 This will search each locally-installed package for names or descriptions
 that contains <string>.
 .SH HANDLING CONFIG FILES
-pacman uses the same logic as rpm to determine action against files
+pacman uses a logic similar to rpm to determine action against files
 that are designated to be backed up.  During an upgrade, it uses 3
 md5 hashes for each backup file to determine the required action:
 one for the original file installed, one for the new file that's about
@@ -210,9 +210,9 @@
 The new one is identical to the current one.  Win win.  Install the new file.
 .TP
 original=\fBX\fP, current=\fBY\fP, new=\fBZ\fP
-All three files are different.  So we install the new file, but back up the
-old one to a .pacsave extension.  This way the user can move the old configuration
-file back into place if he wishes.
+All three files are different.  So we extract the new new file with a .pacnew
+extension, and keep the old one. This way the user can overwrite the old
+configuration with the new one if he wishes.
 .SH CONFIGURATION
 pacman will attempt to read \fI/etc/pacman.conf\fP each time it is invoked.  This
 configuration file is divided into sections or \fIrepositories\fP.  Each section
